docker_cli/_docker_cuda.py

```diff
@@ -0,0 +1,284 @@
+            """
+            ).strip()
+        )
+        body += "\n" + self.generate_install_lines(
+            build_targets=self._runtime_build_targets
+        )
+        return body
+
+    def generate_dev_dockerfile(self) -> str:
+        """
+        Generates a dockerfile for the CUDA Dev image.
+
+        Returns
+        -------
+        str
+            The generated dockerfile body.
+        """
+        body = self.generate_install_lines(build_targets=self._dev_build_targets)
+
+        body = "USER root\n" + body
+
+        version_name = "${CUDA_VERSION_MAJOR}.${CUDA_VERSION_MINOR}"
+        body += (
+            "\n"
+            + textwrap.dedent(
+                f"""\
+                ENV CUDAHOSTCXX=x86_64-conda-linux-gnu-c++
+                ENV CUDACXX=/usr/local/cuda-{version_name}/bin/nvcc
+
+                USER $DEFAULT_USER
+                """
+            ).strip()
+        )
+        return body
+
+    @abstractmethod
+    def _generate_initial_lines(self, repo_ver: str, *, arch: str = "x86_64") -> str:
+        """
+        Generates the first lines of the cuda install dockerfile.
+
+        These lines typically acquire the necessary CUDA repo and configure the
+        image to install CUDA.
+
+        Returns
+        -------
+        str
+            The initial lines of the CUDA install dockerfile.
+        """
+        ...
+
+    @abstractmethod
+    def generate_install_lines(self, build_targets: Iterable[str]) -> str:
+        """
+        Generates the install lines for the cuda install dockerfile.
+
+        These lines typically use the packet manager to install the given CUDA
+        libraries and then clean the packet manager cache.
+
+        Returns
+        -------
+        str
+            The install lines of the CUDA install dockerfile.
+        """
+        ...
+
+    @property
+    def package_manager(self) -> PackageManager:
+        """
+        The package manager used by this dockerfile generator.
+
+        This ensures that all subclasses of this class will have a package manager
+        property, but doesn't require the abstract base class to specify it.
+
+        Raises
+        ------
+        AttributeError
+            If no package manager is present on the object.
+        """
+        if not hasattr(self, "_package_manager"):
+            raise AttributeError(  # pragma: no cover
+                "Attribute 'package_manager' not found."
+            )
+        return getattr(self, "_package_manager")
+
+
+class AptGetCUDADockerfileGen(CUDADockerfileGenerator):
+    """
+    A CUDA dockerfile generator for debian-based apt-get systems
+    """
+
+    _dev_build_targets = [
+        "cuda-cudart-dev-$CUDA_PKG_VERSION",
+        "cuda-nvcc-$CUDA_PKG_VERSION",
+        "libcufft-dev-$CUDA_PKG_VERSION",
+    ]
+
+    def __init__(self, url_reader: URLReader):
+        super(AptGetCUDADockerfileGen, self).__init__(url_reader=url_reader)
```

Comment:
In Python3, you don't need to supply arguments to `super()` (unless you're using multiple inheritance).

https://stackoverflow.com/a/10483204

```suggestion
        super().__init__(url_reader=url_reader)
```